flake.nix

```diff
@@ -158,19 +176,15 @@
           };
         };
         checks = {
-          shellCheck = pkgs.runCommand "shellcheck"
-            { nativeBuildInputs = [ pkgs.shellcheck ]; }
-            ''
-              shellcheck ${packages.exampleInstallerFiles}/install.sh
-              touch $out
-            '';
+          treefmt = (treefmt-nix.lib.evalModule pkgs treefmt-config).config.build.check self;
```

Comment:
Why is this called `evalModule`? What's the module that's being evaluated? And why is the check in `config.build`? (I know these are more questions for how `treefmt-nix` is designed, but it seems that there's some knowledge that I'm missing to understand why this code has to be written like this, and I'm hesitant to merge this without understanding what's going on.)

Reply:
`treefmt-nix` is a library that configure treefmt using nixos modules. The `evalModule` gather the different nixos module options available for the different formaters and apply our custom configuration (`treefmt-config`) to return an attribute set containing the [validated configuration](https://github.com/numtide/treefmt-nix/blob/main/module-options.nix#L195) and [different outputs](https://github.com/numtide/treefmt-nix/blob/main/module-options.nix#L86). [One of these outputs](https://github.com/numtide/treefmt-nix/blob/main/module-options.nix#L146) is `check` which is a flake check to validate that treefmt doesn't reformat anything.

Note that there is a [flake module](https://github.com/numtide/treefmt-nix/blob/main/flake-module.nix) that hides this but it works only with [flake parts](https://github.com/hercules-ci/flake-parts) and we are currently using `flake-utils`.